askama_derive/src/parser/expr.rs

```diff
@@ -139,13 +135,14 @@ fn expr_char_lit(i: &str) -> IResult<&str, Expr<'_>> {
     map(char_lit, Expr::CharLit)(i)
 }
 
-fn expr_var(i: &str) -> IResult<&str, Expr<'_>> {
-    map(identifier, Expr::Var)(i)
-}
-
-fn expr_path(i: &str) -> IResult<&str, Expr<'_>> {
-    let (i, path) = path(i)?;
-    Ok((i, Expr::Path(path)))
+fn expr_path_or_var_or_bool_lit(i: &str) -> IResult<&str, Expr<'_>> {
+    map(path_or_identifier, |v| match v {
+        super::PathOrIdentifier::Path(v) => Expr::Path(v),
+        super::PathOrIdentifier::Identifier(v) => match v {
```

Comment:
I think we could outline this into three separate top-levels arms instead of a nested match? That seems nicer to me.